src/main/java/io/fabric8/jenkins/openshiftsync/JobProcessor.java

```diff
@@ -109,14 +111,38 @@ private void populateNamespaceFolder(Jenkins activeInstance, ItemGroup parent, S
 		}
 	}
 
+	private BuildConfigProjectProperty updateUID(BuildConfigProjectProperty buildConfigProjectProperty, BuildConfigProjectProperty newProperty){
+    String UID = buildConfigProjectProperty.getUid();
+    String newUID = newProperty.getUid();
+    if (!UID.contains(DISABLE_PRUNE_PREFIX)){
+      buildConfigProjectProperty.setUid(newUID);
+    }
+    if (isJobPruningDisabled(buildConfigProjectProperty) && !UID.contains(DISABLE_PRUNE_PREFIX)) {
+      buildConfigProjectProperty.setUid(DISABLE_PRUNE_PREFIX+UID);
+      logger.info("Prune disabled on Jenkins Job " + jenkinsJobFullName(buildConfig));
+    }
+
+    String oldUID = UID.replace(DISABLE_PRUNE_PREFIX,"");
+    String oldName = buildConfigProjectProperty.getNamespace() + "/" + buildConfigProjectProperty.getNamespace() + "-" + buildConfigProjectProperty.getName();
+    System.out.println("-------------- Comparing names "+jenkinsJobFullName(buildConfig)+"------------ "+oldName);
+    System.out.println("-------------- Comparing UID -------"+UID+"--------"+oldUID+"--------"+newUID);
+
+    if (UID.contains(DISABLE_PRUNE_PREFIX) && !oldUID.equals(newUID) && oldName.equals(jenkinsJobFullName(buildConfig))){
```

Comment:
I am not sure that we need to check that the prefix is present here: In case of the a first synchronisation, we should be able to detect at the "first pass" that the name is the same but uid differences, and the annotation is present, so, we override the uid ?


and this also applies with a second sync, if the prefix is here, we should also try to sync and update the uid if relevant. 
Am I correct?